components/Subscribe/subscribeButton.tsx

```diff
@@ -0,0 +1,22 @@
+import { FC } from "react";
+
+interface SubscribeButtonProps {
+  setIsModalOpen: (isOpen: boolean) => void;
+  className?: string;
+}
+
+export const SubscribeButton: FC<SubscribeButtonProps> = ({ setIsModalOpen, className }) => {
+  const handleClick = () => {
```

Comment:
More of the same logic for settting isModalOpen(true), and setting the localStorage item. See in Subscribe/index.tsx also have multiple functions or logic that handles the localstorage and modal boolean.

Reply:
Maybe all of them are neccessary for it to work I dont know, you explain to me if that is the case so I can understand

Reply:
if I remove the setting of localStorage from button functionality , the local storage won't update correctly based on click. For example, when I close the modal the value is set to false in local storage, and on click , yes, it will show the modal again, but the local storage won't be updated to true, and that is messing up my blur logic since its checking for the local storage value and changing accordingly.